### PR TITLE
[1LP][RFR] Automated quota test for memory, storage and vms

### PR DESCRIPTION
**Pull request by akarol**
Adding tests for feature quota to cover cases for memory,storage and vm quota.
{{pytest: -v -k 'test_tenant_quota_max_memory_check or test_tenant_quota_max_storage_check or test_tenant_quota_max_num_vms_check'}}

### CODE REVIEW

**Comment by akarol** [comment]
Its failing in 5.7z due to INTERNALERROR SproutException

File: cfme/tests/infrastructure/test_infra_quota.py [hunk 1]
```diff
@@ -238,3 +237,113 @@ def test_tenant_quota_max_cpu_check(
...
+        * Apart from the usual provisioning settings, set memory greater then tenant quota memory.
+        * Submit the provisioning request and wait for it to finish.
+        * Visit the requests page. The last message should state quota validation message.
+
+    Metadata:
+        test_flag: provision
+    """
+    memory_data = {'memory_cb': True, 'memory': 2}
+    roottenant = Tenant.get_root_tenant()
+    roottenant.set_quota(**memory_data)
```
Comment (jkrocil) [inline cfme/tests/infrastructure/test_infra_quota.py]: Setting and un-setting the quota shouldn't be done within a test but from a yield_fixture. In addition, I don't see where we unset the quota, which could potentially cause weird issues with any provisioning tests ran after this one. I see that there is a yield fixture with cleanup created for the 'test_group_\*' tests in this module but not for the 'test_tenant_\*' ones which all seem to operate within the root tenant.

We definitely need to add cleanup to all of the 'test_tenant_*' tests.

File: cfme/tests/infrastructure/test_infra_quota.py [hunk 2]
```diff
@@ -238,3 +237,113 @@ def test_tenant_quota_max_cpu_check(
...
+        * Apart from the usual provisioning settings, set storage greater then tenant quota storage.
+        * Submit the provisioning request and wait for it to finish.
+        * Visit the requests page. The last message should state quota validation message.
+
+    Metadata:
+        test_flag: provision
+    """
+    storage_data = {'storage_cb': True, 'storage': 1}
+    roottenant = Tenant.get_root_tenant()
+    roottenant.set_quota(**storage_data)
```
Comment (jkrocil) [inline cfme/tests/infrastructure/test_infra_quota.py]: Same here ^

File: cfme/tests/infrastructure/test_infra_quota.py [hunk 3]
```diff
@@ -238,3 +237,113 @@ def test_tenant_quota_max_cpu_check(
...
+        * Submit the provisioning request.
+        * Approve the request and wait for it to finish.
+        * Visit the requests page. The last message should state quota validation message.
+
+    Metadata:
+        test_flag: provision
+    """
+    vm_data = {'vm_cb': True, 'vm': 1}
+    roottenant = Tenant.get_root_tenant()
+    roottenant.set_quota(**vm_data)
```
Comment (jkrocil) [inline cfme/tests/infrastructure/test_infra_quota.py]: Same here ^

**Comment by jkrocil** [comment]
I also initiated a retest on PRT.

File: cfme/tests/infrastructure/test_infra_quota.py (hunk 1 above)
Comment (akarol) [inline cfme/tests/infrastructure/test_infra_quota.py]: Added the fixture for setting and resetting the tenant quota.

**Comment by psav** [comment]
So much DRY worthiness here.....but happy to wait on it until later, RFR should not be set unless failures are explained

**Comment by akarol** [comment]
@psav  oops sorry,I missed one step from the RFR process.
So I have 1 test case "test_tenant_quota_max_num_vms_check" failing in upstream,which is actually passing when i run it locally.

#py.test test_infra_quota.py -k test_tenant_quota_max_num_vms_check
Loading templates from trackerbot                                                                                                                                                                                  
no test annotation found in None
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.11, pytest-2.9.2, py-1.4.32, pluggy-0.3.1
rootdir: /home/akarol/workspace/cfme_tests, inifile: 
plugins: wait-for-1.0.8
collected 6 items 
collected 4 items (uncollected 2 items)
Appliance's streams: [upstream]
test_infra_quota.py .

======================================================== 3 tests deselected by "-ktest_tenant_quota_max_num_vms_check -m 'not long_running and not perf'" =========================================================
==================================================================================== 1 passed, 3 deselected in 485.29 seconds =====================================================================================
collecting artifacts

File: cfme/tests/infrastructure/test_infra_quota.py [hunk 4]
```diff
@@ -88,6 +88,46 @@ def set_group_cpu():
     group.remove_tag("Quota - Max CPUs *", "2")
 
 
+@pytest.yield_fixture(scope="module")
+def set_tenant_cpu():
+    cpu_data = {'cpu_cb': True, 'cpu': 2}
+    reset_cpu_data = {'cpu_cb': False}
+    roottenant = Tenant.get_root_tenant()
```
Comment (akarol) [inline cfme/tests/infrastructure/test_infra_quota.py]: @izapolsk Agreed. will make it more readable.

**Pull request merged**